File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> counts;
        
        for (int i=0; i<n; ++i) {
            ++counts[nums[i]];
            if (counts[nums[i]] > n/2) {
                return nums[i];
            }
        }
        
        return 0;
    }
};

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    const count = {};
    let n = nums.length;
    
    for (let i=0; i<n; ++i) {
        count[nums[i]] = count[nums[i]] == null ? 1 : ++ count[nums[i]];
        if (count[nums[i]] > Math.floor(n/2)) 
            return nums[i];
    }
    
    return 0;
};